lib/pg_ha_migrations/hacks/add_index_on_only.rb

```diff
@@ -0,0 +1,33 @@
+require "active_record/connection_adapters/postgresql_adapter"
+require "active_record/connection_adapters/postgresql/schema_creation"
+
+module PgHaMigrations
+  module ActiveRecordHacks
+    module IndexAlgorithms
+      def index_algorithms
+        super.merge(only: "ONLY")
+      end
+    end
+
+    module CreateIndexDefinition
+      def visit_CreateIndexDefinition(o)
+        if o.algorithm == "ONLY"
+          o.algorithm = nil
+
+          original_query = super
+
+          if original_query.gsub(" ON ").count > 1
```

Comment:
The original idea was to prevent a really obscure situation where you have a table or column name like `"foo ON bar"` (and there's a test for that), but agree it could maybe be too restrictive.  We could maybe replace `ON <table name>` with `ON ONLY <table name>`?

Reply:
This seems a bit better to me: https://github.com/braintree/pg_ha_migrations/pull/85/commits/22e74dab8ff40965bbe4be515122c4419dc7ab91

Reply:
Do we know that this has to start with `CREATE INDEX[ CONCURRENTLY] `? Could we anchor to the front of string with that?

Reply:
You would also need to account for `UNIQUE` as well. I think what I have currently is pretty robust. It would be incredibly rare for `<index name> ON <table name>` to show up more than once in the query string. And even if it does, it would probably be in a where clause somewhere. Since we're doing `sub` and not `gsub` it would correctly replace the first occurrence at the start of the string

Reply:
Oh, good point on "first occurrence".